hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -127,7 +127,7 @@ public HoodieTimeline filterPendingReplaceTimeline() {
   @Override
   public HoodieTimeline filterPendingCompactionTimeline() {
     return new HoodieDefaultTimeline(
-        instants.stream().filter(s -> s.getAction().equals(HoodieTimeline.COMPACTION_ACTION)), details);
+        instants.stream().filter(s -> s.getAction().equals(HoodieTimeline.COMPACTION_ACTION) && !s.isCompleted()), details);
```

Comment:
In fact when action equals to `COMPACTION_ACTION`, it means the compaction is pending(requested or inflight), since the state of COMPACTION is `xxx.compaction.requested -> xxx.compaction.inflight -> xxx.commit`, so i think it should be ok to filter pending compaction here, however we would add comment for this.

Reply:
yes makes sense to add the check (although, if the code is consistent with current behavior, do we need this fix?. it might be misleading?). I am happy to side with you both :)

can we add a unit test around this?

Reply:
> do we need this fix
We need this fix to keep the correct semantics.

You can not assume that any INFLIGHT compaction instants would finally be committed successfully. Although you might rolls back the legacy compaction but at least we should keep the method do the right thing.